onnxruntime/core/framework/tensor_type_and_shape.cc

```diff
@@ -136,9 +136,9 @@ ORT_API(enum ONNXType, OrtGetValueType, _In_ const OrtValue* value) {
   try {
     auto v = reinterpret_cast<const ::onnxruntime::MLValue*>(value);
     onnxruntime::MLDataType type = v->Type();
-    OrtTypeInfo* out;
+    OrtTypeInfo* out = nullptr;
     OrtStatus* ptr = OrtTypeInfo::FromDataTypeImpl(type, nullptr, nullptr, &out);
-    if (ptr != nullptr) {
+    if (ptr != nullptr || out == nullptr) {
```

Comment:
If you hit the 'out == nullptr' isn't ptr a nullptr and you shouldn't call OrtReleaseStatus on it?

Reply:
OrtReleaseStatus accepts nullptr, like the 'free' function

Reply:
OK. Do we need the additional check on out == nullptr to avoid the warning? If not I think it unecessarily complicates things as checking the returned status ptr should be sufficient to know if it failed. If it isn't, FromDataTypeImpl should be fixed to ensure it is.

Reply:
It is needed. Because the compiler(GCC) don't know OrtTypeInfo::FromDataTypeImpl will set 'out' pointer to a valid value whenever the function returns nullptr.

Reply:
The error is:
```
/home/chasun/src/onnxruntime/onnxruntime/core/framework/tensor_type_and_shape.cc:139:18: error: ‘out’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
     OrtTypeInfo* out;
                  ^
lto1: all warnings being treated as errors
```